packages/medusa/src/loaders/helpers/routing/index.ts

```diff
@@ -222,30 +295,41 @@ export class RoutesLoader {
           }
 
           /**
-           * If the developer has not exported the authenticate flag
-           * we default to true.
+           * If the developer has not exported the
+           * AUTHENTICATE flag we default to true.
            */
           const shouldRequireAuth =
             import_[AUTHTHENTICATE] !== undefined
               ? (import_[AUTHTHENTICATE] as boolean)
               : true
 
-          if (
-            shouldRequireAuth &&
-            absolutePath.includes(join("api", "admin"))
-          ) {
-            config.shouldRequireAdminAuth = shouldRequireAuth
-          }
+          /**
+           * If the developer has not exported the
+           * CORS flag we default to true.
+           */
+          const shouldAddCors =
+            import_["CORS"] !== undefined ? (import_["CORS"] as boolean) : true
 
-          if (
-            shouldRequireAuth &&
-            absolutePath.includes(join("api", "store", "me"))
-          ) {
-            config.shouldRequireCustomerAuth = shouldRequireAuth
+          if (route.startsWith("/admin")) {
+            if (shouldAddCors) {
+              config.shouldAppendAdminCors = true
+            }
+
+            if (shouldRequireAuth) {
+              config.shouldRequireAdminAuth = true
+            }
           }
 
-          if (absolutePath.includes(join("api", "store"))) {
+          if (route.startsWith("/store")) {
             config.shouldAppendCustomer = true
+
+            if (shouldAddCors) {
+              config.shouldAppendStoreCors = true
+            }
+          }
+
+          if (shouldRequireAuth && route.startsWith("/store/me")) {
```

Comment:
question: Don't we have other customer-facing routes that require authentication?

Reply:
The safeguard is there as we only want to apply required customer authentication on custom `/store/me/*` endpoints, out of the box. So here we check if the endpoint is a `me` endpoint, and whether or not the user has opted out authentication via `export const AUTHENTICATE = false`. If both checks are valid then we apply the authentication middleware.

Reply:
shouldn't we always apply the auth on /store/me or if the flag is enabled?

Reply:
Yes, and that is also what we do - if its a store/me route + shouldRequireAuth (eg. const AUTENTICATE !== false) then we set the flag which results in the middleware being applied

Reply:
We always apply CORS on /store/me/*, unless the user has explicitly opted out of authentication by setting `export const AUTHENTICATE = false`.

The AUTHENTICATE flag functions as follows:

If the custom endpoint is within /admin/* or /store/me/*, we add the respective authentication middleware by default. If the endpoint's path is outside these, then we don't apply any authentication middleware, even if `export const AUTHENTICATE = true` is declared.

The flag's primary role is to provide an opt-out mechanism from Medusa's default authentication scheme, where `/admin/*` routes are protected by admin authentication middleware, and `/store/me/*` routes by customer authentication middleware. Should users have a special use case that necessitates a `/admin` or `/store/me` endpoint without authentication, they can disable it by setting `export const AUTHENTICATE = false` in the corresponding `route.ts` file.

This design ensures that users can easily align with the core Medusa route patterns with no additional action required. For instance, to add a new admin endpoint, simply create a `route.ts` in the `/api/admin` directory, and we handle the rest. The same principle applies to customer-specific endpoints.

However, if users intend to introduce new endpoints to their storefront API that require customer authentication, they must apply the middleware themselves. For example, to establish endpoints for memberships, one would create `/api/store/membership/**/route.ts`, and then include the following in `middlewares.ts`:

```
export const config = {
  routes: [
    {
      matcher: "/store/membership/*",
      middlewares: [requireCustomerAuthentication()]
    }
  ]
}
```

Reply:
but with the current condition, you can opt out from the required auth for /store/me, wouldn't it be a security breach since it returns the user personal information? shouldn't we always apply the auth on those?

Also, if a user want to make all store end point authenticated? would it mean I can simply do
```ts
export const config = {
  routes: [
    {
      matcher: "/store/*",
      middlewares: [requireCustomerAuthentication()]
    }
  ]
}
```

Reply:
Yes that's what I am saying I think we are talking past each other 😅 

I don't know what the use case would be, and yes it would be a security breach, if the endpoint returns any personal information. But it's not different from if you used the "old" way of registering endpoints and you forgot to add `requireCustomerAuth` (or just opted not to do so) to your custom `/store/me` endpoints. With this approach it's added per default, **unless** you explicitly decide to opt-out of authentication. 

I don't think there should ever be a case where you choose to opt-out of it, but having the option to do so gives you the same flexibility as with the old express way of registering endpoints, while still being able to provide a sane default of adding authentication to /store/me routes. And with this approach you can't simply forget to add authentication to these types of endpoints, you have to actively opt-out of authentication, so it's actually more safe than our old way of registering endpoints.

And yes if you want to add authentication to every custom store route you have in your project or plugin, then you can do as you have in your example.